hippoclient/collections.py

```diff
@@ -272,3 +274,60 @@ def uncache(client: Client, cache: MultiCache, id: str) -> None:
 
     for product in collection.products:
         uncache_product(client, cache, product.id)
+
+
+def set_collection_visibility(
+    client: Client, collection_id: str, visibility: str
+) -> None:
+    """
+    Update the visibility of all products in a collection.
+
+    Arguments
+    ---------
+    client: Client
+        The client to use for interacting with the hippo API.
+    collection_id : str
+        The ID of the collection whose products' visibility is being updated.
+    visibility : str
+        The new visibility level ('public', 'collaboration', or 'private').
+
+    Raises
+    ------
+    ValueError
+        If an invalid visibility level is provided.
+    httpx.HTTPStatusError
+        If a request to the API fails.
+    """
+
+    # Validate visibility
+    try:
+        visibility_enum = Visibility(visibility)
+    except ValueError:
+        raise ValueError(
+            "Invalid visibility level. Choose from 'public', 'collaboration', or 'private'."
+        )
+
+    # Fetch the collection details
+    collection = read(client, collection_id)
+
+    # Update visibility for each product in the collection
+    console.print(
+        f"Updating visibility to '{visibility}' for all products in collection '{collection.name}'...",
+        style="blue",
+    )
+    for product in collection.products:
+        try:
+            set_product_visibility(client, product.id, visibility_enum.value)
+            console.print(
+                f"Successfully updated visibility for product {product.id}.",
+                style="green",
+            )
+        except Exception as e:
+            console.print(
+                f"Failed to update product {product.id}: {str(e)}", style="red"
+            )
```

Comment:
Never blind catch exceptions, you must always catch specific types (e.g. httpx.HTTPStatusError)